synapse/http/server.py

```diff
@@ -507,14 +507,14 @@ class RootOptionsRedirectResource(OptionsResource, RootRedirect):
 
 
 def respond_with_json(
-    request,
-    code,
-    json_object,
-    send_cors=False,
-    response_code_message=None,
-    pretty_print=False,
-    canonical_json=True,
-):
+    request: Request,
+    code: int,
+    json_object: Any,
+    send_cors: bool = False,
+    response_code_message: Optional[str] = None,
+    pretty_print: bool = False,
+    canonical_json: bool = True,
+) -> NOT_DONE_YET:
```

Comment:
Looks like this function also returns `None` below

Reply:
So I ended up removing this since `NOT_DONE_YET` is a constant, not a type and...mypy somewhat gets confused if you try to put `Optional[]` around it or anything. I added back some comments too.